game-server/data/handlers/ai/instance/empyreanCrucible/VanktristAI.java

```diff
@@ -0,0 +1,144 @@
+package ai.instance.empyreanCrucible;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.aionemu.commons.utils.Rnd;
+import com.aionemu.gameserver.ai.AIName;
+import com.aionemu.gameserver.ai.HpPhases;
+import com.aionemu.gameserver.model.gameobjects.Creature;
+import com.aionemu.gameserver.model.gameobjects.Npc;
+import com.aionemu.gameserver.model.gameobjects.player.Player;
+import com.aionemu.gameserver.utils.PacketSendUtility;
+import com.aionemu.gameserver.utils.ThreadPoolManager;
+import com.aionemu.gameserver.world.WorldPosition;
+
+import ai.AggressiveNoLootNpcAI;
+
+/**
+ * @author w4terbomb
+ */
+@AIName("vanktrist")
+public class VanktristAI extends AggressiveNoLootNpcAI implements HpPhases.PhaseHandler {
+
+	private final HpPhases hpPhases = new HpPhases(75, 50);
+
+	private Future<?> phaseTask;
+	private final AtomicBoolean isAggred = new AtomicBoolean(false);
+	private final AtomicBoolean isStartedEvent = new AtomicBoolean(false);
+
+	private VanktristAI(Npc owner) {
+		super(owner);
+	}
+
+	@Override
+	protected void handleAttack(Creature creature) {
+		super.handleAttack(creature);
+		if (isAggred.compareAndSet(false, true)) {
+			PacketSendUtility.broadcastMessage(getOwner(), 0);
+		}
+		checkPercentage(getLifeStats().getHpPercentage());
+	}
+
+	public void checkPercentage(int hpPercentage) {
+		switch (hpPercentage) {
+			case 50, 75 -> {
+				if (isStartedEvent.compareAndSet(false, true))
+					startPhaseTask();
+			}
+		}
+	}
+
+	@Override
+	public void handleHpPhase(int phaseHpPercent) {
+		switch (phaseHpPercent) {
+			case 75 -> PacketSendUtility.broadcastMessage(getOwner(), 1500210, 0, 0);
+			case 50 -> PacketSendUtility.broadcastMessage(getOwner(), 1500211, 0, 0);
+		}
+	}
+
+	private void startPhaseTask() {
+		phaseTask = ThreadPoolManager.getInstance().scheduleAtFixedRate(() -> {
+			if (isDead()) {
+				cancelPhaseTask();
+			} else {
+				getOwner().queueSkill(19567, 46, 0); //Gravitational Shift.
+				List<Player> players = getLifedPlayers();
+				if (!players.isEmpty()) {
+					int size = players.size();
+					int count = Rnd.get(1, size);
+					for (int i = 0; i < count; i++) {
+						spawnWeakenedDimensionalVortex(players.get(Rnd.get(players.getFirst().getObjectId(), players.size())));
```

Comment:
This will certainly throw an exception, because of the objectId being anything between 0 and INT_MAX.
I'm also not rly sure, what you wanna achieve here. Guess, shuffling and slicing the list would be clearer. You can also do this inside your `getLifedPlayers()` method.

PS: `getLivingPlayers()` instead of `getLifedPlayers()`